test_program/catalog.h

```diff
@@ -2,6 +2,13 @@
 #include<string.h>  
 #include<stdlib.h> 
 #include<unistd.h>
+#include <malloc.h>
+#ifdef UNIT_TESTING
+/* Must be last include */
```

Comment:
must be included last.  And normally included in the *.c file, not in the *.h file.